### PR TITLE
kubeadm: remove kubeadm config print-defaults

### PR DESCRIPTION
**Pull request by rosti**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Remove the old and deprecated `kubeadm config print-defaults` command.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
refs kubernetes/kubeadm#1152

**Special notes for your reviewer**:

This one is for 1.14.

/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/area kubeadm
/assign @fabriziopandini
/assign @timothysc

**Does this PR introduce a user-facing change?**:

```release-note
kubeadm: remove deprecated kubeadm config print-defaults command
```

### CODE REVIEW

**Review by timothysc — APPROVED** [review]
/approve 
/lgtm 

I'm kind of surprised/saddened that there were not tests associated with it.

**Comment by xiangpengzhao** [comment]
`pull-kubernetes-verify` job says:

```
I1128 01:49:10.759] < docs/man/man1/kubeadm-config-print-default.1
I1128 01:49:10.759] List of generated docs doesn't match a freshly built list. Please run hack/update-generated-docs.sh
```

**Comment by xiangpengzhao** [comment]
/hold
@rosti feel free to hold cancel after you fix the test failure :)

**Comment by rosti** [comment]
Thanks @xiangpengzhao ! Should be OK now.
/hold cancel

**Review by yagonobre — APPROVED** [review]
/lgtm

File: cmd/kubeadm/app/cmd/config.go [hunk 1]
```diff
@@ -142,76 +141,31 @@ func runConfigPrintActionDefaults(out io.Writer, componentConfigs []string, conf
...
+	}
 
-	default:
-		// Is this a component config?
-		registration, ok := componentconfigs.Known[componentconfigs.RegistrationKind(apiObject)]
-		if !ok {
-			return []byte{}, errors.Errorf("--api-object needs to be one of %v", getAllAPIObjectNames())
-		}
-		return getDefaultComponentConfigBytes(registration)
+	realobj, ok := registration.GetFromInternalConfig(&defaultedInitConfig.ClusterConfiguration)
```
[the remark this answers is not in the document]
Comment (rosti) [reply cmd/kubeadm/app/cmd/config.go]: Thanks for spotting this one @yagonobre ! Just fixed it.

**Review by luxas — APPROVED** [review]
nit: I think you can delete the `// TODO: Probably move to getDefaultedClusterConfig?` comment now.

/lgtm
/approve

Thanks @rosti for the cleanup!

**Pull request merged**